pyrate/core/dem_error.py

```diff
@@ -43,18 +43,22 @@ def remove_dem_error(ifgs: List, params: dict) -> None:
     ifg0 = Ifg(ifg0_path)
     ifg0.open(readonly=True)
 
-    # calculate per-pixel lon/lat
-    lon, lat = get_lonlat_coords(ifg0)
-    # calculate per-pixel radar coordinates
-    az, rg = get_radar_coords(ifg0, params)
-    # calculate per-pixel look angle (also calculates and saves indcidence and azimuth angles)
-    look_angle = calc_local_geometry(ifg0, ifg0_path, az, rg, lon, lat, params)
-    # loop over all Ifgs in stack
-    ifgs = [shared.Ifg(p) for p in ifg_paths] if isinstance(ifgs[0], str) else ifgs
-    process_ifgs = mpiops.array_split(ifgs)
-    for ifg in process_ifgs:
-        bperp = calc_local_baseline(ifg, az, look_angle, params)
-    # todo: add code to invert for DEM error using the per-pixel baseline values and the unwrapped phase
+    # not currently implemented for ROIPAC data which breaks some tests
+    # if statement can be deleted once ROIPAC is deprecated from PyRate
+    if not ifg0.meta_data[ifc.PYRATE_INSAR_PROCESSOR] == 'ROIPAC':
+
+      # calculate per-pixel lon/lat
+      lon, lat = get_lonlat_coords(ifg0)
+      # calculate per-pixel radar coordinates
+      az, rg = get_radar_coords(ifg0, params)
+      # calculate per-pixel look angle (also calculates and saves indcidence and azimuth angles)
+      look_angle = calc_local_geometry(ifg0, ifg0_path, az, rg, lon, lat, params)
+      # loop over all Ifgs in stack
+      ifgs = [shared.Ifg(p) for p in ifg_paths] if isinstance(ifgs[0], str) else ifgs
+      process_ifgs = mpiops.array_split(ifgs)
+      for ifg in process_ifgs:
+          bperp = calc_local_baseline(ifg, az, look_angle, params)
```

Comment:
So this is implementing parallelism by sending each ifg to a process to be done independently? What's the performance like for full-size data? Is it worth tiling-up the per pixel calculations?

Reply:
To be tested once I've implemented the actual DEM error correction.